darshan-runtime/lib/darshan-ldms.c

```diff
@@ -0,0 +1,249 @@
+/*
+ *   Copyright (C) 2016 Intel Corporation.
+ *   See COPYRIGHT notice in top-level directory.
+ *    
+ */
+
+#ifdef HAVE_CONFIG_H
+# include <darshan-runtime-config.h>
+#endif
+
+#define _XOPEN_SOURCE 500
+#define _GNU_SOURCE
+#include <netdb.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Check for LDMS libraries if Darshan is built --with-ldms */
+#ifdef HAVE_LDMS
+#include <ldms/ldms.h>
+#include <ldms/ldmsd_stream.h>
+#include <ovis_util/util.h>
+#include "ovis_json/ovis_json.h"
+#include "darshan-ldms.h"
+
+struct darshanConnector dC = {
+     .ldms_darsh = NULL,
+     .ldms_lib = 0,
+     };
+
+ldms_t ldms_g;
+static void event_cb(ldms_t x, ldms_xprt_event_t e, void *cb_arg)
+{
+        switch (e->type) {
+        case LDMS_XPRT_EVENT_CONNECTED:
+                sem_post(&dC.conn_sem);
+                dC.conn_status = 0;
+                break;
+        case LDMS_XPRT_EVENT_REJECTED:
+                ldms_xprt_put(x);
+                dC.conn_status = ECONNREFUSED;
+                break;
+        case LDMS_XPRT_EVENT_DISCONNECTED:
+                ldms_xprt_put(x);
+                dC.conn_status = ENOTCONN;
+                break;
+        case LDMS_XPRT_EVENT_ERROR:
+                dC.conn_status = ECONNREFUSED;
+                break;
+        case LDMS_XPRT_EVENT_RECV:
+                sem_post(&dC.recv_sem);
+                break;
+        case LDMS_XPRT_EVENT_SEND_COMPLETE:
+                break;
+        default:
+                printf("Received invalid event type %d\n", e->type);
+        }
+}
+
+#define SLURM_NOTIFY_TIMEOUT 5
```

Comment:
This doesn't appear to be used, so we could probably drop? At first glance, I wondered if there was something Slurm specific about the code in this file but that doesn't appear to be the case, so could also avoid some confusion.

Reply:
Yep, I agree. It's not being used anywhere --- will drop it.